pennylane/gradients/general_shift_rules.py

```diff
@@ -13,69 +13,109 @@
 # limitations under the License.
 """Contains a function for generating generalized parameter shift rules."""
 import functools
+import itertools
 import warnings
+
 import numpy as np
 import pennylane as qml
 
 
-@functools.lru_cache(maxsize=None)
-def get_shift_rule(frequencies, shifts=None):
-    r"""Computes the parameter shift rule for a unitary based on its generator's eigenvalue frequency
-    spectrum.
+def _process_gradient_recipe(gradient_recipe, tol=1e-10):
+    """Utility function to process gradient recipes.

```

Comment:
I still find it sometimes confusing that "gradient recipe" refers just to shift terms, it sounds like it is a strategy like "adjoint" or "param_shift"... :) 

But no need to change,! If you like, you could define in the docstring what data structure a gradient recipe is?

Reply:
oh I think I will change the name :) `_process_shifts` maybe